pkg/rehearse/jobs.go

```diff
@@ -158,85 +187,141 @@ func filterJob(source *prowconfig.Presubmit, allowVolumes bool) error {
-	var rehearsal prowconfig.Presubmit
-	deepcopy.Copy(&rehearsal, job)
-	for _, container := range rehearsal.Spec.Containers {
-		for index := range container.Env {
-			env := &(container.Env[index])
-			if env.ValueFrom == nil {
-				continue
+func inlineCiOpConfig(container v1.Container, ciopConfigs config.CompoundCiopConfig, loggers Loggers) error {
+	for index := range container.Env {
+		env := &(container.Env[index])
+		if env.ValueFrom == nil {
+			continue
+		}
+		if env.ValueFrom.ConfigMapKeyRef == nil {
+			continue
+		}
+		if config.IsCiopConfigCM(env.ValueFrom.ConfigMapKeyRef.Name) {
+			filename := env.ValueFrom.ConfigMapKeyRef.Key
+
+			loggers.Debug.WithField(logCiopConfigFile, filename).Debug("Rehearsal job uses ci-operator config ConfigMap, needed content will be inlined")
+
+			ciopConfig, ok := ciopConfigs[filename]
+			if !ok {
+				return fmt.Errorf("ci-operator config file %s was not found", filename)
 			}
-			if env.ValueFrom.ConfigMapKeyRef == nil {
-				continue
+
+			ciOpConfigContent, err := yaml.Marshal(ciopConfig)
+			if err != nil {
+				loggers.Job.WithError(err).Error("Failed to marshal ci-operator config file")
+				return err
 			}
-			if config.IsCiopConfigCM(env.ValueFrom.ConfigMapKeyRef.Name) {
-				filename := env.ValueFrom.ConfigMapKeyRef.Key
 
-				logFields := logrus.Fields{logCiopConfigFile: filename, logCiopConfigRepo: targetRepo, logRehearsalJob: job.Name}
-				loggers.Debug.WithFields(logFields).Debug("Rehearsal job uses ci-operator config ConfigMap, needed content will be inlined")
+			env.Value = string(ciOpConfigContent)
+			env.ValueFrom = nil
+		}
+	}
+	return nil
+}
 
-				ciopConfig, ok := ciopConfigs[filename]
-				if !ok {
-					return nil, fmt.Errorf("ci-operator config file %s was not found", filename)
-				}
+// JobConfigurer ...
+type JobConfigurer struct {
+	presubmits config.Presubmits
+	periodics  []prowconfig.Periodic
 
-				ciOpConfigContent, err := yaml.Marshal(ciopConfig)
-				if err != nil {
-					loggers.Job.WithError(err).Error("Failed to marshal ci-operator config file")
-					return nil, err
-				}
+	ciopConfigs config.CompoundCiopConfig
+	templates   []config.ConfigMapSource
+	profiles    []config.ConfigMapSource
 
-				env.Value = string(ciOpConfigContent)
-				env.ValueFrom = nil
-			}
-		}
+	prNumber     int
+	loggers      Loggers
+	allowVolumes bool
+	templateMap  map[string]string
+}
+
+// NewJobConfigurer ...
+func NewJobConfigurer(presubmits config.Presubmits, periodics []prowconfig.Periodic, ciopConfigs config.CompoundCiopConfig, prNumber int, loggers Loggers, allowVolumes bool, templates []config.ConfigMapSource, profiles []config.ConfigMapSource) *JobConfigurer {
+	return &JobConfigurer{
+		presubmits:   filterPresubmits(presubmits, allowVolumes, loggers.Job),
+		periodics:    filterPeriodics(periodics, allowVolumes, loggers.Job),
```

Comment:
These are only necessary in the corresponding `Configure*Rehearsal` method, they can be parameters instead of members.

Reply:
You are right, but I would like to keep them here for now to make the API more extensible and avoid more refactoring in the future.